packages/web-app-files/src/components/AppBar/AppBar.vue

```diff
@@ -97,8 +98,8 @@
               </ul>
             </oc-drop>
           </template>
-          <size-info v-if="selectedFiles.length > 0" class="oc-mr-s uk-visible@l" />
-          <batch-actions />
+          <size-info v-if="hasBulkActions && selectedFiles.length > 0" class="oc-mr uk-visible@l" />
+          <batch-actions v-if="hasBulkActions" />
```

Comment:
Unit tests compain and I'm not sure why batch.actions don't also get a `&& selectedFiles.length > 0` condition...I'll add it to make them pass (and test acceptance tests in CI)

Reply:
The batch actions are allowed without selected files now - there is a new action mixin in this PR (empty trash bin) which is only visible when no files are selected (on selection there is `delete` action mixin which handles permanent deletion of individually selected files). All action mixins now check if they are enabled with respect to the number of selected resources (some only available for 1 selected resource, some only available for >1 selected resources, some for >0 selected resources, one (empty trash) for 0 selected resources). With that the batch actions always need to be allowed (only can be suppressed with the route meta attribute `hasBulkActions`).

Long story short: we'll need to adapt the unit and integration tests accordingly.